z2/docs/deposit.md

```diff
@@ -22,21 +23,26 @@ Usage: z2 deposit --chain <CHAIN_NAME> --public-key <BLS_PUBLIC_KEY> --peer-id <
+* `--signing-address <SIGNING_ADDRESS>`: Specifies the address which signs cross-chain events.
+* `--deposit-auth-signature <DEPOSIT_AUTH_SIGNATURE>`: The BLS proof-of-possession signature of the validator node.
 
 **Note**: The `--private-key` parameter should be the private key of a wallet that has secured a minimum stake amount of 10 million ZILs.
 
 ### Generating Required Values
-To generate the `public-key`, `pop-signature` and `peer-id`, use the following command inside the zq2 folder. Please pass `PRIVATE_KEY_OF_VALIDATOR` to the command input.
+To generate the `public-key`, `deposit-auth-signature` and `peer-id`, use the following command inside the zq2 folder. Please pass `PRIVATE_KEY_OF_VALIDATOR` and `CHAIN_ID` to the command input. 
+
+By default this tool signs over the address derived from the given secret key to generate the `deposit-auth-signature`, you may want to override this for example if deploying via a Delegation contract. To override pass in `"address": "<CONTROL_ADDRESS>"`.
+
 ```bash
-echo '{"secret_key":"<PRIVATE_KEY_OF_VALIDATOR>"}' | cargo run --bin convert-key
+echo '{"secret_key":"<PRIVATE_KEY_OF_VALIDATOR>", "chain_id": <CHAIN_ID>}' | cargo run --bin convert-key
 ```
 #### Sample run
 ```bash
-$ echo '{"secret_key":"96252e38af375be21d9eb30a6b88abc3836acecaeb2240731fb42e0299e14419"}' | cargo run --bin convert-key
+$ echo '{"secret_key":"96252e38af375be21d9eb30a6b88abc3836acecaeb2240731fb42e0299e14419", "chain_id": 33469}' | cargo run --bin convert-key
     Finished `dev` profile [unoptimized + debuginfo] target(s) in 0.67s
      Running `target/debug/convert-key`
-{"address":"0x3946f9872247af2eb4fe44c81c463e801925b8d4","bls_pop_signature":"90dbc73ac9f40b75acd8279a50447f5f4e61c377f4147d1dc4b6f139b84d5c48180aae67c76db3abc4c205e4c48df5160a88ae1b84ed4869c9660458d7feb5100ccc499dbd4f2131b5a90e34261ae6ea9246abf8c32b75b1f16e63a66eec2214","bls_public_key":"825124961d51c99816848875fa505b75f2e62e69937fe9bfa5fa97711845abd667f05bdc3756f7dba6b7e9e0467a3804","peer_id":"12D3KooWGu8PBoj6vMPafnhA2P7sLumSV1NhQJZ2W2AGiBgc5ATW","tx_pubkey":{"Ecdsa":["3056301006072A8648CE3D020106052B8104000A03420004B7C457DC36C75EADA5675629F1CE0FA93534FB76ADFC49840CC050AE2995FC87764AEB8975D049D19FDA6BFF2B3FF51608034A3FC6708F476A0C9306BA5CBE14",true]}}
+{"address":"0x3946f9872247af2eb4fe44c81c463e801925b8d4","deposit_auth_signature":"a53efd8bad058e4e551b7e9681613a278b782acfe05fb98d536bc95029278704adbb85891cdc7fa384ab7d5008fdd42c0ea70404ab4ec07bcf5e738c92b2be88debdc33014852ead1d9976fcbf7760043615ba74f36181fc87db21f8f8997a44","bls_public_key":"825124961d51c99816848875fa505b75f2e62e69937fe9bfa5fa97711845abd667f05bdc3756f7dba6b7e9e0467a3804","peer_id":"12D3KooWGu8PBoj6vMPafnhA2P7sLumSV1NhQJZ2W2AGiBgc5ATW","tx_pubkey":{"Ecdsa":["3056301006072A8648CE3D020106052B8104000A03420004B7C457DC36C75EADA5675629F1CE0FA93534FB76ADFC49840CC050AE2995FC87764AEB8975D049D19FDA6BFF2B3FF51608034A3FC6708F476A0C9306BA5CBE14",true]}}
```

Comment:
from this example looks like the address in output to the `convert key` is also the one in input to the signing address, is this correct?

Can we add an example also using an address as input?

@DrZoltanFazekas @JamesHinshelwood

Reply:
That's correct, since `z2 deposit` in the example uses the same `PRIVATE_KEY_OF_VALIDATOR` as `convert-key` so the address computed by `convert-key` is the one that z2 will send the deposit transaction from